app/map/page.tsx

```diff
@@ -0,0 +1,91 @@
+"use client";
+import { useState, useEffect } from "react";
+import { Button, Flex } from "antd";
+import { HomeOutlined, ReloadOutlined } from "@ant-design/icons";
+import { logError } from "@/app/utils/logger";
+import { CategoryObject } from "../types/ruleInfo";
+import { getAllRuleData, getBRERules } from "../utils/api";
+import RuleRelationsGraph from "../components/RuleRelationsDisplay/RuleRelationsDisplay";
+import styles from "./map.module.css";
+
+export default function Map() {
+  const [isLoading, setIsLoading] = useState(true);
+  const [klammRules, setKlammRules] = useState<any[]>([]);
+  const [categories, setCategories] = useState<CategoryObject[]>([]);
+  const [location, setLocation] = useState<Location>();
+
+  useEffect(() => {
+    setLocation(window.location);
+  }, []);
+
+  const getOrRefreshRuleList = async () => {
```

Comment:
A similar method to this is used 3 times. Could we put it somewhere so it's not repeated?